StudyHub/StudyHub.BLL/Services/UserService.cs

```diff
@@ -48,6 +59,48 @@ public async Task<UserDTO> UpdateUserAsync(Guid userId, UpdateUserDTO dto)
         return _mapper.Map<UserDTO>(user);
     }
 
+    public async Task<AvatarResponse> UploadAvatarAsync(Guid userId, IFormFile avatar)
+    {
+        var contentPath = _env.ContentRootPath;
+        var path = Path.Combine(contentPath, _avatarConfig.Folder);
+
+        if (!Directory.Exists(path))
+            Directory.CreateDirectory(path);
+
+        var ext = Path.GetExtension(avatar.FileName);
+
+        if (!_avatarConfig.FileExtensions.Contains(ext))
+            throw new IncorrectParametersException("Invalid file extension");
+
+        var newFileName = $"{userId}{ext}";
+        var filePath = Path.Combine(path, newFileName);
+
+        using var stream = new FileStream(filePath, FileMode.Create);
+        await avatar.CopyToAsync(stream);
```

Comment:
this will override files with same name, better to store them in different directories e.g. uploads/users/{userId}/avatar_{unique-suffix}.png

Reply:
the user must have only one avatar. It might be better to add that when the user adds a new avatar, the old avatar is deleted